aries_cloudagent/utils/json.py

```diff
@@ -0,0 +1,100 @@
+
+        The `orjson` library does not introduce whitespace between keys and values,
+        nor after commas. The default behavior of `json.dumps`, however, does include
+        this whitespace. For compatibility purposes, this method reintroduces such
+        whitespace. Although this introduces some overhead, the overall process
+        remains faster than using standard `json.dumps`.
+
+        A capture group in a regular expression is used to cater to the cases where
+        the value following a colon (:) in a JSON string is either a string
+        (starting  with "), a digit, a JSON or a List object (starting with { or [),
+        or a boolean value or null (starting with "t", "f", or "n" respectively).
+
+        This regular expression only operates under the assumption that all keys in
+        the JSON string are strings enclosed in quotes (") which is the default
+        behavior for `json.dumps` and `orjson.dumps`.
+
+        Args:
+            json_str: The compact json string to be formatted.
+
+        Returns:
+            Formatted json string with a space added where appropriate.
+
+        """
+
+        json_str = re.sub(r',([tfn"\d\{\[])', r", \1", json_str)  # space after comma
+        json_str = re.sub(r'":([tfn"\d\{\[])', r'": \1', json_str)  # space after colon
+        return json_str
+
+    @staticmethod
+    def dumps(obj, *args, **kwargs) -> str:
+        """Convert a Python object into a json string.
+
+        Args:
+            obj: The data to be converted
+            *args: Extra arguments to pass to the orjson.dumps() function
+            **kwargs: Extra keyword arguments to pass to the orjson.dumps() function
+
+        Returns:
+            The json string representation of obj
+
+        """
+
+        if "indent" in kwargs:  # not supported in orjson, and only used in demo logs
+            return json.dumps(obj, *args, **kwargs)
+        else:
+            return JsonUtil.format_json(orjson.dumps(obj, *args, **kwargs).decode())
+
+    @staticmethod
+    def loads(s: str, *args, **kwargs) -> Any:
+        """Parse a JSON string into a Python object.
+
+        Args:
+            s: The JSON string to be parsed
+            *args: Extra arguments to pass to the orjson.loads() function
+            **kwargs: Extra keyword arguments to pass to the orjson.loads() function
+
+        Returns:
+            The Python representation of s
+
+        """
+
+        return orjson.loads(s, *args, **kwargs)
+
+    @staticmethod
+    def load(fp, *args, **kwargs) -> Any:
+        """Parse a JSON file into a Python object.
+
+        Args:
+            fp: a .read()-supporting file-like object containing a JSON document
+            *args: Extra arguments to pass to the json.load() function
+            **kwargs: Extra keyword arguments to pass to the json.load() function
+
+        Returns:
+            The Python representation of fp
+
+        """
+
+        return json.load(fp, *args, **kwargs)
```

Comment:
shouldn't we be using orjson? something like:

```
orjson.loads(fo.read())
```

I realize this is only used in test code (currently) but feels somehow wrong to leave it as `json`.

Reply:
Aah, I noticed the same this morning. Turns out, orjson doesn't have a load method! I think it makes sense to remove it here and just use json.load directly in the one test that depends on it

Reply:
but `orjson.loads` can take file-like objects. so my thought was keep all json code under JsonUtils and update the implementation to use `orjson`.

Reply:
By the way, orjson.loads doesn't work on file-like objects. `loads` expects a string / byte array.

Here's how the json.load method can be replaced with orjson.loads:

Using json.load

```python

import json

with open('file.json', 'r') as f:
    data = json.load(f)
```
Equivalent using orjson.loads

```python

import orjson

with open('file.json', 'rb') as f:
    content = f.read()
    data = orjson.loads(content)
```

Is it fine if I leave `load` out of JsonUtil, and just update the test to use orjson.loads / JsonUtil.loads to read the file?

Reply:
This is maybe me being a completist but I don't think leaving that as `json.load` is correct. Currently it is only used in two places, so you could just as well update the calling code to better match up with `orjson.loads`. But more importantly, the impetus for this PR was to gain speed, which was measured by how much quicker the tests ran. The main use case for that `json.load` is in the BDD tests. Perhaps we will see an even greater improvement if everything is using `orjson`, which also supports indent, so we could use that too.  Of more importance is if we need to flag `orjson`/`json`, we can do it in JsonUtils.  I think if it is ever needed in the runtime code, having it in JsonUtils (hopefully) will make it more apparent to that developer.

Reply:
@usingtechnology So, my idea is just to move the `read_json_data` method to JsonUtil ... With the dependency check, it's better for that to be configured in JsonUtil. Then the demo/bdd_support/ module just uses `JsonUtil.read_json_file`. Does that seem okay to you? I'll push that in a minute

Reply:
> @usingtechnology So, my idea is just to move the `read_json_data` method to JsonUtil ... With the dependency check, it's better for that to be configured in JsonUtil. Then the demo/bdd_support/ module just uses `JsonUtil.read_json_file`. Does that seem okay to you? I'll push that in a minute

That sounds good. I think the bigger thing is what @dbluhm pointed out and that configuration will rely on this being bound to a profile, which means not a utility class but more like a service/provider class.